### ISSUE TITLE
Feature request: Use CAS inside placeholders

### ISSUE THREAD

**Issue by wasalm** [maths/moodle-qtype_stack#697]
I'm trying to use {@ ... @} inside of a placeholder, but it doesn't gets evaulated. Can this be made possible in the next version?

**Comment by wasalm** [comment]
Thanks for the fast response Chris, I want to put the following text inside the placeholder:

`Answer in terms of g, l, r,{@ subst(c, massP) @}, {@ subst(c, massQ) @}, {@ subst(c, speed) @}, {@ subst(c, distance) @}\and {@ subst(c, angleSymbol) @}`

Instead of showing
`Answer in terms of g, l, r,M, m, v, d and alpha`
it displays the raw code.

I am creating a question where I randomize the variable name for the students. That is, some students has to answer in the variables `c=[massP = m, massQ = M, angleSymbol = alpha]` and other students has to answer in the variables  `c=[massP = m_p, massQ = m_q, angleSymbol = psi]`. To present the correct question I need to substutite my internal used variables (massP, massQ, etc) with the variables (M, m, etc.). For this I use the `subst` function in CAS. It works for all fields except for the 'Syntax Hint' field. There it still displays the raw code. 

I can circumvent this by putting the syntax hint inside the question text, but for next time it would be more preferable to put this inside the syntax hint.

**Comment by sangwinc** [comment]
OK, so this is in the syntax hint!?

**Comment by aharjula** [comment]
Ok. an input placeholder or syntax-hint like all the other input options other than the teacher's answer is currently static. This is something that is being worked on (the so-called input2-system) but it is not yet ready for release. Personally, I imagine that it is something coming in STACK 5 time-scale as the replacement of the input/validation portion of the question-model is not exactly a small thing and might not be entirely automatically mappable from the old to the new.

However, in input2 the options related to inputs can have a much more complex structure as well as being able to include CASText (which is the templating language where `{@...@}` works) and CAS-evaluated parameters. The primary reason for input2 is the need for more options for inputs and for those options to allow connecting to parameters in question-variables, so basically exactly what you are asking for. 

So not for the next version, but maybe the next major version, well Chris will decide when that happens and what goes into it. However, there are still multiple other large infrastructure projects coming to fruition that will probably appear in versions starting with 4.

**Comment by wasalm** [comment]
> OK, so this is in the syntax hint!?

Yes it is in the syntax hint. (In HTML it is called a placeholder that is why I refer to that.)

**Comment by wasalm** [comment]
> Ok. an input placeholder or syntax-hint like all the other input options other than the teacher's answer is currently static. This is something that is being worked on (the so-called input2-system) but it is not yet ready for release. Personally, I imagine that it is something coming in STACK 5 time-scale as the replacement of the input/validation portion of the question-model is not exactly a small thing and might not be entirely automatically mappable from the old to the new.
> 
> However, in input2 the options related to inputs can have a much more complex structure as well as being able to include CASText (which is the templating language where `{@...@}` works) and CAS-evaluated parameters. The primary reason for input2 is the need for more options for inputs and for those options to allow connecting to parameters in question-variables, so basically exactly what you are asking for.
> 
> So not for the next version, but maybe the next major version, well Chris will decide when that happens and what goes into it. However, there are still multiple other large infrastructure projects coming to fruition that will probably appear in versions starting with 4.

Thanks for the reply. I will look forward to this new feature.

**Comment by sangwinc** [comment]
The syntax hint is now castext, this will be released in STACK 4.9.0.
